src/coords.c

```diff
@@ -411,9 +429,11 @@ PyObject* GetCoords(PyArrayObject* arr, int include_z) {
   /* We work with a "cursor" that tells the get_coordinates function where
   to write the coordinate data into the output array "result" */
   cursor = 0;
+  geom_i = -1;
```

Comment:
This seems a bit awkward; what about initializing to 0, and then moving the update of it from line 436 to the end of the `do` loop (after line 455, before line 456)

Reply:
I did that first but then I got in trouble with the `continue` on line 443. So the alternative is doing `geom_1++` there also.